### ISSUE TITLE
Attribution dialog links do not work if MapView context is wrapped

### ISSUE THREAD

**Issue by msuonp** [mapbox/mapbox-maps-android#2304]
## Environment

- Android OS version: 14
- Maps SDK Version: 11.1.0

## Observed behavior and steps to reproduce

1. Have a map view in a fragment which has the context wrapped in a `ContextWrapper`. This is the case for example when the fragment is injected with Hilt using `@AndroidEntryPoint` annotation.
2. Click the 'i'-icon on top of the map in bottom-left corner to open the attribution dialog.
3. Click any of the items in the dialog except the telemetry settings.
4. The dialog closes and nothing else happens.

## Expected behavior

The attribution link should open in a browser window.

## Notes / preliminary analysis

The link is not opened, because `AttributionDialogManagerImpl` skips opening the link if the context is not an instance of  `Activity`. See the source [here](https://github.com/mapbox/mapbox-maps-android/blob/7fb578c1eff912fa8e9b93de8542b1726b34048e/plugin-attribution/src/main/java/com/mapbox/maps/plugin/attribution/AttributionDialogManagerImpl.kt#L150).

Instead of comparing the context directly to Activity, I think the dialog manager should try to find an Activity from the context, for example like this:

```
fun Context.getActivity(): Activity? {
    var context = this
    while (context is ContextWrapper) {
        if (context is Activity) {
            return context
        }
        context = context.baseContext
    }
    return null
}
```

**Comment by kiryldz** [comment]
@msuonp sound like a valid issue, we'll take a look and try to fix it before v11.3.0-beta.1 next week.

**Comment by jush** [comment]
Hi @msuonp! @kiryldz did a fix but instead of iterating up through the base context, we just try catch:
```Kotlin
  private fun showWebPage(url: String) {
    try {
      val intent = Intent(Intent.ACTION_VIEW)
      intent.data = Uri.parse(url)
      mapView.context.startActivity(intent)
    } catch (exception: ActivityNotFoundException) { // explicitly handling if the device hasn't have a web browser installed. #8899
      Toast.makeText(...).show()
    } catch (t: Throwable) {
      Toast.makeText(...).show()
    }
  }
```
If users have custom behaviour in the ContextWrapper then we should be able to respect it.

**Comment by msuonp** [comment]
Sounds good, thanks for fixing this!

**Comment by kiryldz** [comment]
@msuonp closing this, feel free to re-open if the issue will persist in v11.3.0-beta.1 that will be out next week.

**Comment by rvanderlinden** [comment]
@jush I have a related issue using Android MapBox 11.3.1 and Android 11. 

Clicking any attribution dialog url displays a toast indicating _"Calling startActivity() from outside of an Activity context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?"_. The url is not opened.

I have tried it with a `MapView` in a fragment layout, or by programmatically instantiating it with the `Activity` as its context param, but both result in the toast mentioned above.

I'm not exactly sure how to proceed. Is this something you can reproduce, or is there a way we could override the default `showWebPage` behaviour?

**Comment by msuonp** [comment]
@rvanderlinden Same result here, attribution URLs are not opened. SDK 11.3.0 and Android 14. I didn't notice this before, because I had a workaround for the earlier SDK.

@jush It seems I can't reopen the issue, because I didn't close it myself.

**Comment by kiryldz** [comment]
@msuonp @rvanderlinden thanks for reporting, I've created an internal issue so this does not get lost. We'll take a look.

**Comment by kiryldz** [comment]
@msuonp @rvanderlinden this will be fixed in v11.4.0-rc.1 coming out later today. I'll close the ticket now, please feel free to re-open (again 😄).
